### ISSUE TITLE
Panic when using shatter tool.

### ISSUE THREAD

**Issue by cloudcalvin** [IvoryDuke/HillVacuum#8]
Hi @IvoryDuke

This work of yours is thoroughly impressive and I have learned a lot from studying it. Thank you.
I will use some of the concepts for my own (non-game) editor. 

How long did it take for you to make this? 

I noticed that there is a panic every time I do the shatter operation. I am not sure if I am doing something wrong, or if its an issue with the shatter tool itself or just the item selection logic.

I have created a pull request for this: 
https://github.com/IvoryDuke/HillVacuum/pull/7

But I could be misdiagnosing the issue.

**Comment by IvoryDuke** [comment]
Thank you for the opened issue I have just finished reviewing it.

If I remember correctly I started HV in November-December 2022 and have been working on it in my free time.
It is my first Rust project so it has taken me a lot of time since I have been learning the language as I added new features.
It is also my first major coding project.

Thank you a lot for your time! seeing people taking interest in it is motivating me to add new features.
